Filter_removed_comments_from_search

tracyspacy opened the pull request:
As proposed in [#4576](https://github.com/LemmyNet/lemmy/issues/4576)

SleeplessOne1917 commented:
Should deleted comments also be filtered out?

Nutomic commented:
This is the wrong place to change it as its also used to get the comments for a post among other things. Its necessary to show removed comments there so they can be restored. As its only about search you should filter them in [search.rs](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/api/search.rs) instead.

tracyspacy commented:
@Nutomic got your point. Added filter for removed comments in search.rs . Deleted comments are still shown in search results, should they remain or not?

dessalines commented on crates/apub/src/api/search.rs:
```diff
@@ -201,6 +201,12 @@ pub async fn search(
+  let comments = comments
+    .into_iter()
+    .filter(|c| !c.comment.removed && !c.comment.deleted)
+    .collect();
```
This isn't the right way to handle it either, filtering needs to be in SQL. 

In the comment query logic, you can check for `search_term` (since if that's provided, you know its a search), and filter out deleted and removed based on that.

dessalines replied:
First try to add that [here](https://github.com/LemmyNet/lemmy/blob/main/crates/db_views/src/comment_view.rs#L219), into the existing `if let Some(search)`  . 

If that fails tho, your approach looks good also.

SleeplessOne1917 commented on crates/apub/src/api/search.rs:
```diff
@@ -201,6 +201,12 @@ pub async fn search(
+  //filtering out removed and deleted comments from search results
+  let comments = comments
+    .into_iter()
+    .filter(|c| !c.comment.removed && !c.comment.deleted)
```
Nitpick: you can DeMorgan this to only use 1 negation.

```suggestion
    .filter(|c| !(c.comment.removed || c.comment.deleted))
```
